mhkit/dolfyn/adp/clean.py

```diff
@@ -157,30 +157,38 @@ def water_depth_from_amplitude(ds, thresh=10, nfilt=None) -> None:
             "Please manually remove 'depth' if it needs to be recalculated."
         )
 
+    # Use "avg" velocty if standard isn't available.
+    # Should not matter which is used.
+    tag = []
+    if hasattr(ds, "vel"):
+        tag += [""]
+    if hasattr(ds, "vel_avg"):
+        tag += ["_avg"]
+
     # This finds the maximum of the echo profile:
-    inds = np.argmax(ds["amp"].values, axis=1)
+    inds = np.argmax(ds["amp" + tag[0]].values, axis=1)
     # This finds the first point that increases (away from the profiler) in
     # the echo profile
-    edf = np.diff(ds["amp"].values.astype(np.int16), axis=1)
+    edf = np.diff(ds["amp" + tag[0]].values.astype(np.int16), axis=1)
     inds2 = (
         np.max(
             (edf < 0)
-            * np.arange(ds["vel"].shape[1] - 1, dtype=np.uint8)[None, :, None],
+            * np.arange(ds["vel" + tag[0]].shape[1] - 1, dtype=np.uint8)[None, :, None],
             axis=1,
         )
         + 1
     )
 
     # Calculate the depth of these quantities
-    d1 = ds["range"].values[inds]
-    d2 = ds["range"].values[inds2]
+    d1 = ds["range" + tag[0]].values[inds]
+    d2 = ds["range" + tag[0]].values[inds2]
     # Combine them:
     D = np.vstack((d1, d2))
     # Take the median value as the estimate of the surface:
     d = np.median(D, axis=0)
 
     # Throw out values that do not increase near the surface by *thresh*
-    for ip in range(ds["vel"].shape[1]):
+    for ip in range(ds["vel" + tag[0]].shape[1]):
         itmp = np.min(inds[:, ip])
         if (edf[itmp:, :, ip] < thresh).all():
             d[ip] = np.nan
```

Comment:
on 188 here you are using the median on d1 and d2 and I notice on 194 you add a nan to the array.

Will d1 or d2 ever have nan? If so median will always return nan. Is that the behavior you want? Or would `np.nanmedian` be preferred?

Reply:
Hmmm this is a Levi function... I'll add nan to all of those median min and max functions, because if this is called after another QC function that would be a problem.